lib/generateClass.js

```diff
@@ -284,14 +307,222 @@ function writeToJson(def, cw) {
+            }
+
+            if(field){
+              // console.log(field.identifier.name);
+              if (!alreadyMappedElements.has(classMethodChain)) {
+                const constraintsLength = field.constraintsFilter.includesType.constraints.length;
+                const slicing = sliceMap[element.path] && sliceMap[element.path][element.sliceName];
+                generateFromFHIRAssignment(field.card.isList, baseIsList, constraintsLength, element, fieldMapPath, classMethodChain, fhirProfile, slicing, cw);
+                alreadyMappedElements.set(classMethodChain, element.path);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  if (fhirExtension) {
+    // extension not a profile
+
+    // When calling a method on an es6 instance that has a FHIR extension, you may either want to method to resolve directly to a value (e.g. 12)
+    //   or you may want it to be represented as an extension if it is not mapped to an element in a FHIR resource, such as 
+    //   [{url: 'http://extension, valueInteger: 12}]
+    //   asExtension tells which way to import it depending where in the parent class this exists (extension or not)
+    cw.bl(`if (asExtension)`, () => {
+      fhirExtension.differential.element.forEach( (element, i) => {
+        if(element.path.startsWith('Extension.value') && element.path !== 'Extension.value[x]'){
+          // Simple extension with a valueType
+          let name = element.path.split('.')[1];
+
+          cw.ln(`inst.value = fhir['${name}'];`);
+        } else if(element.path === 'Extension.extension' && element.max != '0'){
+          // Complex extension
+
+          // The current implementation only seems to reference nested extensions, which promotes reuse of 
+          // components of extensions, instead of expanding them out into a form like Patient.extension.extension.valueSting
+          // If this changes, this code could need to be more generic to traverse the extension tree recursively
+          // Need to figure out the name of the field we are looking at by grabbing the extension and looking at the identifier
+          // This seemed better than parsing the URL, which seems like a somewhat arbitrary format
+          let instance = fhir.extensions.find(e => e.url === element.type[0].profile).identifier[0].value;
+          let methodName = toSymbol(instance.split('.')[instance.split('.').length-1]);
+          // find the right extension in the list
+          const url = element.type[0].profile;
+          const varName = `match_${i}`; // ensure a unique variable name here
+          cw.ln(`const ${varName} = fhir['extension'].find(e => e.url == '${url}');`);
+          cw.bl(`if (${varName} != null)`, () => {
+            cw.ln(`inst.${methodName} = createInstanceFromFHIR('${instance}', ${varName}, true);`); // asExtension = true here, false(default value) everywhere else
+          });
+        }
+      });
+    });
+  }
```

Comment:
I'll be honest with you...  I'm having trouble wrapping my head around the extension stuff above.  I'd recommend heavily testing it, and if it works, I'll trust that it's right! ;-)